lib/modules.nix

```diff
@@ -317,58 +326,58 @@ rec {
-      else
-        let
-          /* Add the modules of the current option to the list of modules
-             already collected.  The options attribute except either a list of
-             submodules or a submodule. For each submodule, we add the file of the
-             current option declaration as the file use for the submodule.  If the
-             submodule defines any filename, then we ignore the enclosing option file. */
-          options' = toList opt.options.options;
-
-          getSubModules = opt.options.type.getSubModules or null;
-          submodules =
-            if getSubModules != null then map (packSubmodule opt._file) getSubModules ++ res.options
-            else if opt.options ? options then map (coerceOption opt._file) options' ++ res.options
-            else res.options;
-        in opt.options // res //
-          { declarations = res.declarations ++ [opt._file];
-            options = submodules;
-          } // typeSet
-    ) { inherit loc; declarations = []; options = []; } opts;
+        in
+          if bothHave "default" || bothHave "example" || bothHave "description" || bothHave "apply" || (bothHave "type" && (! typesMergeable))
+          then
+            throw "The option `${showOption loc}' in `${opt._file}' is already declared in ${showFiles res.declarations}."
+          else
+            let
+              /* Add the modules of the current option to the list of modules
+                 already collected.  The options attribute except either a list of
+                 submodules or a submodule. For each submodule, we add the file of the
+                 current option declaration as the file use for the submodule.  If the
+                 submodule defines any filename, then we ignore the enclosing option file. */
+              options' = toList opt.options.options;
+
+              getSubModules = opt.options.type.getSubModules or null;
+              submodules =
+                if getSubModules != null then map (packSubmodule opt._file) getSubModules ++ res.options
+                else if opt.options ? options then map (coerceOption opt._file) options' ++ res.options
+                else res.options;
+            in opt.options // res //
+              {
+                declarations = res.declarations ++ [ opt._file ];
+                options = submodules;
+              } // typeSet
+      ) { inherit loc; declarations = [ ]; options = [ ]; } opts;
 
   /* Merge all the definitions of an option to produce the final
      config value. */
   evalOptionValue = loc: opt: defs:
     let
       # Add in the default value for this option, if any.
       defs' =
-          (optional (opt ? default)
-            { file = head opt.declarations; value = mkOptionDefault opt.default; }) ++ defs;
+        (optional (opt ? default)
+        { file = head opt.declarations; value = mkOptionDefault opt.default; }
+        ) ++ defs;
```

Comment:
the attribute set should be indented to distinct between the function name and function arguments. the last `)` is already fixed in the latest `nixpkgs-fmt` commit.